src/main/java/yevhen/bookstore/model/Book.java

```diff
@@ -52,4 +56,9 @@ public class Book {
     @EqualsAndHashCode.Exclude
     @ToString.Exclude
     private Set<Category> categories = new HashSet<>();
+
+    @OneToMany(mappedBy = "book", cascade = CascadeType.REMOVE, orphanRemoval = true)
+    @EqualsAndHashCode.Exclude
+    @ToString.Exclude
+    private Set<CartItem> cartItems = new HashSet<>();
```

Comment:
Do you really need this logic? better mark CartItem as unavailable. But it's strange if CartItem just disappears from ShoppingCart

Reply:
i have added an additional check to avoid removal the book from db if this book is still in one of the carts